src/freenet/node/updater/NodeUpdater.java

```diff
@@ -511,8 +511,14 @@ public synchronized boolean canUpdateNow() {
 	/** Called when the fetch URI has changed. No major locks are held by caller. 
 	 * @param uri The new URI. */
 	public void onChangeURI(FreenetURI uri) {
-		kill();
+		kill(); // unsubscribes from the old uri
 		this.URI = uri;
+		try {
```

Comment:
This looks suspiciously like the `start()` method. Maybe we should simply call that method here?

Reply:
That’s a good point. It’s not exactly the same (we won’t want to stop updating when the key is invalid), but very close. I factored out a subscribe() method.